mesonbuild/compilers/fortran.py

```diff
@@ -72,7 +72,7 @@ def sanity_check(self, work_dir_: str, environment: 'Environment') -> None:
         if binary_name.is_file():
             binary_name.unlink()
 
-        source_name.write_text('print *, "Fortran compilation is working."; end', encoding='utf-8')
+        source_name.write_text('print *, "Fortran compilation is working."', encoding='utf-8')
```

Comment:
I know nothing about fortran. What's the meaning of this change?

Reply:
Ah, sorry; actually this one is an incomplete fix for fortran compilers which was more fully fleshed out in #9248; I shouldn't have put this here.

Reply:
Essentially a complete fortran program to test with is:

```fortran
program main
print *, "Hello"
end program
```

Which can be written as `program main; print*, "Fortran compilation is working."; end program`

Depending on the standards; some compilers accept not having `program` in the beginning; and others accept not explicitly requiring `end program`.

Reply:
The other PR is merged now, so you can "put this here" by rebasing. :D